src/app/InteractionModelEngine.cpp

```diff
@@ -485,6 +540,223 @@ void InteractionModelEngine::AddReadClient(ReadClient * apReadClient)
     mpActiveReadClientList = apReadClient;
 }
 
+bool InteractionModelEngine::CheckResourceAndEvictExceededSubscription(FabricIndex aFabricIndex, bool aForceEvict)
```

Comment:
```suggestion
bool InteractionModelEngine::TrimFabricToCompliance(FabricIndex aFabricIndex, bool aForceEvict)
```

Reply:
Infact this is not "Trim", we will only evict at most one subscription per call, to utilize resources and avoid bias ( when choosing the fabric / subscription to evict.

Reply:
Maybe EvictOneSubscriptionForCompliance?

Reply:
That's a little too wordy. I think `TrimFabric` perhaps might be better, since it doesn't mention "how" much trimming is happening.